adapters/ddf-adapter/src/main/java/com/connexta/replication/adapters/ddf/DdfNodeAdapterFactory.java

```diff
@@ -13,32 +13,117 @@
+import com.connexta.replication.adapters.ddf.csw.GetRecordsMessageBodyReader;
 import com.connexta.replication.adapters.ddf.rest.DdfRestClientFactory;
 import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.xml.namespace.QName;
+import net.opengis.cat.csw.v_2_0_2.AcknowledgementType;
+import net.opengis.cat.csw.v_2_0_2.CapabilitiesType;
+import net.opengis.cat.csw.v_2_0_2.GetCapabilitiesType;
+import net.opengis.cat.csw.v_2_0_2.GetRecordsResponseType;
+import net.opengis.cat.csw.v_2_0_2.GetRecordsType;
 import org.codice.ddf.cxf.client.ClientFactoryFactory;
+import org.codice.ddf.cxf.client.SecureCxfClientFactory;
 import org.codice.ditto.replication.api.NodeAdapter;
 import org.codice.ditto.replication.api.NodeAdapterFactory;
 import org.codice.ditto.replication.api.NodeAdapterType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Factory for creating {@link DdfNodeAdapter}s for the {@link
  * org.codice.ditto.replication.api.Replicator}.
  */
 public class DdfNodeAdapterFactory implements NodeAdapterFactory {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(DdfNodeAdapterFactory.class);
+
   private final DdfRestClientFactory ddfRestClientFactory;
 
   private final ClientFactoryFactory clientFactory;
 
+  private final int connectionTimeout;
+
+  private final int receiveTimeout;
+
   public DdfNodeAdapterFactory(
-      DdfRestClientFactory ddfRestClientFactory, ClientFactoryFactory clientFactory) {
-    this.ddfRestClientFactory = ddfRestClientFactory;
+      ClientFactoryFactory clientFactory, int connectionTimeout, int receiveTimeout) {
+    this.connectionTimeout = connectionTimeout;
+    this.receiveTimeout = receiveTimeout;
+    this.ddfRestClientFactory =
+        new DdfRestClientFactory(clientFactory, connectionTimeout, receiveTimeout);
     this.clientFactory = clientFactory;
+    LOGGER.debug(
+        "Created a DdfNodeAdapterFactory with a connection timeout of {} seconds and a receive timeout of {} seconds",
+        connectionTimeout,
+        receiveTimeout);
   }
 
   @Override
   public NodeAdapter create(URL url) {
-    return new DdfNodeAdapter(ddfRestClientFactory, clientFactory, url);
+    SecureCxfClientFactory<Csw> ddfCswClientFactory =
+        clientFactory.getSecureCxfClientFactory(
+            url.toString() + "/csw",
+            Csw.class,
+            initProviders(),
+            null,
+            true,
+            false,
+            connectionTimeout,
+            receiveTimeout);
+    return new DdfNodeAdapter(ddfRestClientFactory, ddfCswClientFactory, url);
+  }
+
+  private List<Object> initProviders() {
+
+    CswJAXBElementProvider getRecordsTypeProvider = new CswJAXBElementProvider<>();
+    getRecordsTypeProvider.setMarshallAsJaxbElement(true);
+    List<String> jaxbElementClassNames = new ArrayList<>();
```

Comment:
✏️ Can you create with a size of 5 based on what you use below? This will avoid allocating more memory than required.